ext/dom/php_dom.c

```diff
@@ -471,9 +481,12 @@ static HashTable* dom_get_debug_info_helper(zend_object *object, int *is_temp) /
 
 		ZEND_ASSERT(string_key != NULL);
 
+		DOM_G(suppress_warnings) = true;
```

Comment:
I would just set this to true outside of the loop and reset it to false after the loop, that's imo a little bit cleaner.

Reply:
Yes, makes total sense.